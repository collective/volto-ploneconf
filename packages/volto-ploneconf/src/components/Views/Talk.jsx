import { isEmpty } from 'lodash';
import {
  Container as SemanticContainer,
  Grid,
  Header,
  Image,
  Label,
  Segment,
} from 'semantic-ui-react';
import { flattenToAppURL } from '@plone/volto/helpers';
import { Component, UniversalLink } from '@plone/volto/components';
import { When } from '@plone/volto/components/theme/View/EventDatesInfo';
import DefaultImageSVG from '@plone/volto/components/manage/Blocks/Listing/default-image.svg';
import config from '@plone/volto/registry';

const TalkView = (props) => {
  const { content } = props;
  const Container =
    config.getComponent({ name: 'Container' }).component || SemanticContainer;
  const color_mapping = {
    beginner: 'green',
    advanced: 'yellow',
    professional: 'purple',
  };
  return (
    <Container id="view-wrapper talk-view">
      <h1 className="documentFirstHeading">
        <span className="type_of_talk">{content.type_of_talk.title}: </span>
        {content.title}
      </h1>
      {content.description && (
        <p className="documentDescription">{content.description}</p>
      )}
      <Segment floated="right">
        {content.image?.scales && (
          <Image
            src={flattenToAppURL(content.image?.scales?.preview?.download)}
            size="medium"
            alt={content.title}
          />
        )}
        {content.start && !content.hide_date && (
          <>
            <Header dividing sub>
              When
            </Header>
            <When
              start={content.start}
              end={content.end}
              whole_day={content.whole_day}
              open_end={content.open_end}
            />
          </>
        )}
        {content.room && (
          <>
            <Header dividing sub>
              Where
            </Header>
            <p>{content.room.title}</p>
          </>
        )}
        {content.audience && (
          <>
            <Header dividing sub>
              Audience
            </Header>
            {content.audience?.map((item) => {
              let color = color_mapping[item.token] || 'green';
              return (
                <Label key={item.token} color={color}>
                  {item.title}
                </Label>
              );
            })}
          </>
        )}
      </Segment>
      <div dangerouslySetInnerHTML={{ __html: content.details.data }} />
      <Segment clearing>
        <Header dividing>Speaker</Header>
        {content.speaker?.length > 0 && (
          <Grid>
            <Grid.Row columns={5}>
              {content.speaker.map((el) => (
                <Grid.Column>
                  <UniversalLink href={el['@id']}>
                    {isEmpty(el.image_scales) ? (
                      <img
                        src={
                          config.getComponent({
                            name: 'DefaultImage',
                            dependencies: ['listing', 'summary'],
                          }).component || DefaultImageSVG
                        }
                        alt={el.title}
                        className="ui image"
                      />
                    ) : (
                      <Component
                        componentName="PreviewImage"
                        item={el}
                        image_field={isEmpty(el.image_scales) ? null : 'image'}
                        showDefault={true}
                        alt={el.title}
                        responsive={true}
                        className="ui image"
                      />
                    )}
                    <div>{el.title}</div>
                  </UniversalLink>
                </Grid.Column>
              ))}
            </Grid.Row>
          </Grid>
        )}
      </Segment>
    </Container>
  );
};
export default TalkView;
