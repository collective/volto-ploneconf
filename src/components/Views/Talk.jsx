import React from 'react';
import { flattenToAppURL } from '@plone/volto/helpers';
import { Container, Header, Image, Label, Segment } from 'semantic-ui-react';
import { When } from '@plone/volto/components/theme/View/EventDatesInfo';

const TalkView = ({ content }) => {
  const color_mapping = {
    Beginner: 'green',
    Advanced: 'yellow',
    Professional: 'purple',
  };

  return (
    <Container id="view-wrapper talk-view">
      <h1 className="documentFirstHeading">
        {content.type_of_talk.title || content.type_of_talk.token}:{' '}
        {content.title}
      </h1>
      <Segment floated="right">
        {content.start && !content.hide_date && (
          <>
            <Header dividing sub>
              When
            </Header>
            <When
              start={content.start}
              end={content.end}
              whole_day={content.whole_day}
              open_end={content.open_end}
            />
          </>
        )}
        {content.audience && (
          <>
            <Header dividing sub>
              Audience
            </Header>
            <div>
              {content.audience?.map((item) => {
                let audience = item.title;
                let color = color_mapping[audience] || 'green';
                return (
                  <Label key={audience} color={color} tag>
                    {audience}
                  </Label>
                );
              })}
            </div>
          </>
        )}
      </Segment>
      {content.description && (
        <p className="documentDescription">{content.description}</p>
      )}
      {content.details && (
        <div dangerouslySetInnerHTML={{ __html: content.details.data }} />
      )}
      <Segment clearing>
        {content.speaker && <h3>{content.speaker}</h3>}
        {content.website ? (
          <p>
            <a href={content.website}>{content.company}</a>
          </p>
        ) : (
          <p>{content.company}</p>
        )}
        {content.email && (
          <p>
            Email: <a href={`mailto:${content.email}`}>{content.email}</a>
          </p>
        )}
        {content.twitter && (
          <p>
            Twitter:{' '}
            <a href={`https://twitter.com/${content.twitter}`}>
              {content.twitter.startsWith('@')
                ? content.twitter
                : '@' + content.twitter}
            </a>
          </p>
        )}
        {content.github && (
          <p>
            Github:{' '}
            <a href={`https://github.com/${content.github}`}>
              {content.github}
            </a>
          </p>
        )}
        {content.image && (
          <Image
            src={flattenToAppURL(content.image.scales.preview.download)}
            size="small"
            floated="right"
            alt={content.image_caption}
            avatar
          />
        )}
        {content.speaker_biography && (
          <div
            dangerouslySetInnerHTML={{
              __html: content.speaker_biography.data,
            }}
          />
        )}
      </Segment>
    </Container>
  );
};
export default TalkView;
