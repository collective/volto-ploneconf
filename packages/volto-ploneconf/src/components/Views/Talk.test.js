import renderer from 'react-test-renderer';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';
import Talk from './Talk';
const mockStore = configureStore();

const store = mockStore({
  intl: {
    locale: 'en',
    messages: {},
  },
});

test('renders a talk view component with only required props', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Talk
        content={{
          title: 'Security of Plone',
          description: 'What makes Plone secure?',
          type_of_talk: { title: 'Talk', token: 'talk' },
          details: {
            'content-type': 'text/html',
            data: '<p>some details about this <strong>talk</strong>.</p>',
            encoding: 'utf8',
          },
        }}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
