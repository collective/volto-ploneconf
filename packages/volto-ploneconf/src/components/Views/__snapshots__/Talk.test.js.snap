// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a talk view component with only required props 1`] = `
<div
  className="ui container"
  id="view-wrapper talk-view"
>
  <h1
    className="documentFirstHeading"
  >
    <span
      className="type_of_talk"
    >
      Talk
      : 
    </span>
    Security of Plone
  </h1>
  <p
    className="documentDescription"
  >
    What makes Plone secure?
  </p>
  <div
    className="ui right floated segment"
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>some details about this <strong>talk</strong>.</p>",
      }
    }
  />
  <div
    className="ui clearing segment"
  >
    <div
      className="ui dividing header"
    >
      Speaker
    </div>
  </div>
</div>
`;
